{
    "assignee": null,
    "assignees": [],
    "body": "The following is legitimate CPython code:\n\n```\nd = {}\nworks = { k : [n for n in names] for k, names in d.items() }\nfails = { k : set(n for n in names) for k, names in d.items() }\n```\n\nThe \"fails\" line causes the following error:\ncython_defect.py:3:34: local variable 'names' referenced before assignment\n\nThis may have the same root cause as #1351 (based purely on the description).\n",
    "closed_at": "2017-02-07T06:36:00Z",
    "comment_data": [
        {
            "body": "Another failure that looks related is below (cython v0.25.2). The entire code is in the error message:\r\n\r\n```\r\nError compiling Cython file:\r\n------------------------------------------------------------\r\n...\r\nclass C(object):\r\n  a = {}\r\n  b = list(k for k in a)\r\n                      ^\r\n------------------------------------------------------------\r\n\r\n/tmp/x.py:3:23: undeclared name not builtin: a\r\n```",
            "created_at": "2017-01-24T20:50:54Z",
            "html_url": "https://github.com/cython/cython/issues/1479#issuecomment-274934395",
            "id": 274934395,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/1479",
            "updated_at": "2017-01-24T20:50:54Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/274934395",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/7066873?v=3",
                "events_url": "https://api.github.com/users/shalabhc/events{/privacy}",
                "followers_url": "https://api.github.com/users/shalabhc/followers",
                "following_url": "https://api.github.com/users/shalabhc/following{/other_user}",
                "gists_url": "https://api.github.com/users/shalabhc/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/shalabhc",
                "id": 7066873,
                "login": "shalabhc",
                "organizations_url": "https://api.github.com/users/shalabhc/orgs",
                "received_events_url": "https://api.github.com/users/shalabhc/received_events",
                "repos_url": "https://api.github.com/users/shalabhc/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/shalabhc/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/shalabhc/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/shalabhc"
            }
        },
        {
            "body": "Duplicate of #1351.",
            "created_at": "2017-02-07T06:36:00Z",
            "html_url": "https://github.com/cython/cython/issues/1479#issuecomment-277914137",
            "id": 277914137,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/1479",
            "updated_at": "2017-02-07T06:36:00Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/277914137",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/491659?v=3",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            }
        }
    ],
    "comments": 2,
    "comments_url": "https://api.github.com/repos/cython/cython/issues/1479/comments",
    "created_at": "2016-10-07T19:29:47Z",
    "event_data": [
        {
            "actor": {
                "avatar_url": "https://avatars.githubusercontent.com/u/491659?v=3",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2017-02-07T06:35:55Z",
            "event": "labeled",
            "id": 951556291,
            "label": {
                "color": "444444",
                "name": "R: duplicate"
            },
            "url": "https://api.github.com/repos/cython/cython/issues/events/951556291"
        },
        {
            "actor": {
                "avatar_url": "https://avatars.githubusercontent.com/u/491659?v=3",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            },
            "commit_id": null,
            "commit_url": null,
            "created_at": "2017-02-07T06:36:00Z",
            "event": "closed",
            "id": 951556375,
            "url": "https://api.github.com/repos/cython/cython/issues/events/951556375"
        }
    ],
    "events_url": "https://api.github.com/repos/cython/cython/issues/1479/events",
    "html_url": "https://github.com/cython/cython/issues/1479",
    "id": 181749774,
    "labels": [
        {
            "color": "444444",
            "default": false,
            "id": 425556894,
            "name": "R: duplicate",
            "url": "https://api.github.com/repos/cython/cython/labels/R:%20duplicate"
        }
    ],
    "labels_url": "https://api.github.com/repos/cython/cython/issues/1479/labels{/name}",
    "locked": false,
    "milestone": null,
    "number": 1479,
    "repository_url": "https://api.github.com/repos/cython/cython",
    "state": "closed",
    "title": "Cython compilation fails for valid nested generator comprehension.",
    "updated_at": "2017-02-07T06:36:00Z",
    "url": "https://api.github.com/repos/cython/cython/issues/1479",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/406380?v=3",
        "events_url": "https://api.github.com/users/gwk/events{/privacy}",
        "followers_url": "https://api.github.com/users/gwk/followers",
        "following_url": "https://api.github.com/users/gwk/following{/other_user}",
        "gists_url": "https://api.github.com/users/gwk/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/gwk",
        "id": 406380,
        "login": "gwk",
        "organizations_url": "https://api.github.com/users/gwk/orgs",
        "received_events_url": "https://api.github.com/users/gwk/received_events",
        "repos_url": "https://api.github.com/users/gwk/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/gwk/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/gwk/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/gwk"
    }
}